app/src/main/java/com/pavesid/androidacademy/ui/fragments/MoviesDetailsFragment.kt

```diff
@@ -0,0 +1,115 @@
+package com.pavesid.androidacademy.ui.fragments
+
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import coil.load
+import com.pavesid.androidacademy.R
+import com.pavesid.androidacademy.data.local.FakeRepository
+import com.pavesid.androidacademy.databinding.FragmentMoviesDetailsBinding
+import com.pavesid.androidacademy.ui.activities.MainActivity
+import com.pavesid.androidacademy.ui.adapters.CastAdapter
+import com.pavesid.androidacademy.ui.decorations.CastItemDecoration
+import com.pavesid.androidacademy.utils.Utils
+
+class MoviesDetailsFragment : Fragment() {
+
+    private var _binding: FragmentMoviesDetailsBinding? = null
+    private val binding: FragmentMoviesDetailsBinding
+        get() = _binding!!
+
+    private val castAdapter by lazy { CastAdapter() }
+
+    private var addTitle = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            addTitle = it.getString(PARAM_TITLE, "")
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentMoviesDetailsBinding.inflate(layoutInflater)
+
+        initView()
+
+        return binding.root
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        _binding = null
+    }
+
+    private fun initView() {
+
+        val movie = FakeRepository.getEndGameMovie()
+
+        if (movie.image == "") {
+            binding.detailsOrig.load(Drawable.createFromStream(requireActivity().assets.open("orig.png"), null))
+        } else {
+            binding.detailsOrig.load(movie.image)
+        }
+
+        binding.detailsName.apply {
+            text = resources.getString(R.string.name_custom, movie.title, addTitle)
+            paint.shader = Utils.getShaderForGradientTextView(this)
```

Comment:
in Kotlin it is better practice to use extension functions instead of static functions